phac-nml/irida-next
# Attachments Concatenation UI

## What does this PR do and why?
A user can select two or more files of the same type and concatenate them together.
Fixes #286.

## Screenshots or screen recordings
![Screenshot from 2023-11-15 13-57-37](https://github.com/phac-nml/irida-next/assets/325703/2bf76938-ba6b-42a2-98ae-d4c43274d35c)

## How to set up and validate locally
A. Keeping the original files.
1. Navigate to sample.
2. Upload two or more files of the same type.
3. Check the files to be concatenated.
4. Click the `Concatenate Files` button.
5. Fill in a base name.
6. Click the `Concatenate` button.
7. Verify the original & concatenated file.

B. Removing the original files.
1. Navigate to sample.
2. Upload two or more files of the same type.
3. Check the files to be concatenated.
4. Click the `Concatenate Files` button.
5. Fill in a base name.
6. Check `Delete originals`.
7. Click the `Concatenate` button.
8. Verify the original & concatenated file.


## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
